cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -132,7 +144,9 @@ protected static String defaultBootstrapServers(String cluster) {
     }
 
     public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
-        CruiseControl cruiseControl = null;
+        CruiseControl cruiseControl = new CruiseControl(kafkaAssembly.getMetadata().getNamespace(),
```

Comment:
why are you initializing the `cruiseControl` instance here? we need that only if it's in the `Kafka.spec`, or? I remember I fixed something like that because deploying the `Kafka` cluster without CC wasn't working anymore.

Reply:
Yea, I added the initialization here to avoid a NPE in this test [1] (but forgot to delete the original initialization after the conditional statement). But you are right, it's safer to initialize the `cruiseControl` var to null here and update the test[1] to handle it properly. I'll update this to how it was before.

[1] https://github.com/kyguy/strimzi-kafka-operator/blob/cc3bb1935dfa03a12dcb734c25be9c845bf8a934/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java#L314-L322

Reply:
Fixed this back to the way it was and updated the tests for your review

Reply:
Changed this back and updated the tests for your review